app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -512,39 +521,45 @@ private void loadJson(BufferedReader in) {
 					item.addLink(new Link(event.getString("signup_url")));
 				}
 
-				JSONObject microlocation = event.getJSONObject("microlocation");
-				String location = microlocation.getString("name");
 
-				if ((line = tentMap.get(location)) == null) {
-					line = new Schedule.Line(location, location);
-					tents.add(line);
-					tentMap.put(location, line);
-				}
+                JSONObject microlocation = event.getJSONObject("microlocation");
+                String location = microlocation.getString("name");
+
+                if ((line = tentMap.get(location)) == null) {
+                    line = new Schedule.Line(location, location);
+                    tents.add(line);
+                    tentMap.put(location, line);
+                }
+
 
 				//Getting value (latitude and longitude) from the map by key (name)
-				String locString = locs.get(line.getTitle());
-				String latitude = locString.substring(0, locString.indexOf(','));
-				String longitude = locString.substring(locString.indexOf(',') + 1);
 
-				//Adding location details here
-				String latlon = null;
-				try {
-					latlon = ("geo:0,0?q=" + longitude + "," +
-							latitude + "(" +
-							URLEncoder.encode(line.getTitle(), "utf-8") + ")");
-				} catch (UnsupportedEncodingException e) {
-					e.printStackTrace();
-				}
+                if(hasMicrolocs) {
+                    String locString = locs.get(line.getTitle());
```

Comment:
@Wilm0r If think there would be any misspelled name the match would be that misspelled name. It depends on organiser when the event is created on our server, but I think this works fine.
Anyways it'd great if you could suggest any other way.

I have made other changes from the reviews.